components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -322,12 +318,77 @@ func (backend *ESClient) InsertInspecProfile(ctx context.Context, data *relaxtin
 	return err
 }
 
-func (backend *ESClient) InsertComplianceRunInfo(ctx context.Context, nodeId string, runDateTime time.Time) error {
+func (backend *ESClient) InsertComplianceRunInfo(ctx context.Context, report *relaxting.ESInSpecReport, runDateTime time.Time) error {
 	mapping := mappings.ComplianceRunInfo
-	err := backend.upsertComplianceRunInfo(ctx, mapping, nodeId, runDateTime)
+	runInfoCh := make(chan relaxting.ESComplianceRunInfo)
+	cErr := make(chan error)
+	firstRunInfoCh := make(chan string)
+
+	go MapReportToRunInfo(report, runDateTime, runInfoCh)
+	go backend.getDocFromNodeRunInfoFromNodeId(ctx, report.NodeID, mappings.ComplianceRunInfo.Index, firstRunInfoCh, cErr)
+
+	runInfo := <-runInfoCh
+	firstRun := <-firstRunInfoCh
+	err := <-cErr
```

Comment:
This wont work incase of success scenarios, the control will wait here till it got a record, and since there is no error, it will keep on wait infinite time. 

please use for loop and select as below
for i := 0; i < 2; i++ {
		select {
		case runInfo := <-runInfoCh:
		case firstRun := <-firstRunInfoCh:
		case err := <-cErr:
		}
	}